### PR TITLE
Update Cloud IoT Gateways Python 3

### PR DESCRIPTION
**Pull request by gguuss**
* Adds support for Python 3 (primarily in print statements)
* Moves response logic out of MQTT callbacks into main function
* Subscribes Gateway to error topic
* Adds a few additional event messages reported to Google Cloud

### CODE REVIEW

**Comment by gguuss** [comment]
I have verified the sample to work on a Raspberry Pi 3 in Python 2 and Python 3.

File: tutorials/cloud-iot-gateways-rpi/cloudiot_mqtt_gateway.py [hunk 1]
```diff
@@ -32,13 +33,18 @@
 ADDR = (HOST, PORT)
 
 udpSerSock = socket.socket(socket.AF_INET, socket.SOCK_DGRAM)
-udpSerSock.setblocking(0)
+udpSerSock.setblocking(False)
 udpSerSock.bind(ADDR)
 
+skip_next_sub = False
```
Comment (hongalex) [inline tutorials/cloud-iot-gateways-rpi/cloudiot_mqtt_gateway.py]: Does this do anything? I can't seem to see where it's being used.

Comment (gguuss) [reply tutorials/cloud-iot-gateways-rpi/cloudiot_mqtt_gateway.py]: This is unused, I'll fix :)

**Comment by ToddKopriva** [comment]
updated document live:
https://cloud.google.com/community/tutorials/cloud-iot-gateways-rpi

**Pull request merged**